movie/src/test/java/com/example/presentation/ui/popular/PopularMoviesViewModelTest.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.presentation.ui.popular
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.example.domain.entity.Movie
+import com.example.domain.interactor.Failure
+import com.example.domain.interactor.movies.GetPopularMoviesUseCase
+import com.example.presentation.ui.util.collectNewValues
+import io.mockk.coEvery
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class PopularMoviesViewModelTest {
+
+    lateinit var popularMoviesViewModel: PopularMoviesViewModel
+
+    @get:Rule
+    val rule = InstantTaskExecutorRule()
+
+    @ExperimentalCoroutinesApi
+    @Before
+    fun init() {
+        Dispatchers.setMain(TestCoroutineDispatcher())
+    }
+
+    @Test
+    fun `when viewmodel is created, the default UiModel is emitted`() {
+        // when
+        popularMoviesViewModel = PopularMoviesViewModel(mockk(), GetPopularMoviesUseCase(mockk()))
+        val default = PopularMoviesUiModel.default.copy(onFetchButtonClicked = popularMoviesViewModel::fetchMovies)
+
+        // then
+        Assert.assertEquals(popularMoviesViewModel.uiModel, default)
+    }
+
+    @Test
+    fun `given a successful network response, when fetchMovies is called, moviesText should be set and isLoading should be toggled`() {
+        // given
+        val movieTitle = "My Movie"
+        val movieOverview = "My Movie Overview"
+        val expectedText = "•Title: My Movie\nOverview: My Movie Overview\n\n"
+
+        val useCase: GetPopularMoviesUseCase = mockk()
+        coEvery { useCase() } returns Result.success(listOf(Movie(movieTitle, movieOverview)))
```

Comment:
`coEvery` is a suspend function - allows mocking on other suspend functions. Mockito doesn't have great support for this